apps/gateway/src/middlewares/ai-models/provider-service.ts

```diff
@@ -0,0 +1,98 @@
+import type {
+  ApiKeyProviderConfig,
+  AwsProviderConfig,
+  GoogleProviderConfig,
+  ProviderConfig,
+  ProviderName,
+} from "@hebo/shared-data/types/providers";
+
+import {
+  getBedrockDefaultConfig,
+  createBedrockProvider,
+  transformBedrockModelId,
+} from "./providers/bedrock";
+import {
+  getGroqDefaultConfig,
+  createGroqProvider,
+  transformGroqModelId,
+} from "./providers/groq";
+import {
+  getVertexDefaultConfig,
+  createVertexProvider,
+  transformVertexModelId,
+} from "./providers/vertex";
+
+import type { Provider as AiProvider } from "ai";
+
+interface Provider {
+  resolveConfig(cfg?: ProviderConfig): Promise<ProviderConfig>;
+  create(cfg?: ProviderConfig): Promise<AiProvider>;
+  resolveModelId(id: string, cfg?: ProviderConfig): Promise<string>;
+}
+
+class BedrockProvider implements Provider {
```

Comment:
You're creating an unnecessary layer of abstraction. The code that is in providers/bedrock.ts should be in this class. This class itself should live in providers/bedrock.ts.